services/user_service/callbacks.py

```diff
@@ -0,0 +1,70 @@
+from constants import database_admin, pubsub_manager, RESULT_TOPIC
+import json
+
+
+def ack_message(message):
+    print(f"Received message : {message}")
+    print(f"Data : {message.data}")
+    message.ack()
+    return message.attributes["message_id"]
+
+
+def get_users_callback(message):
+    message_id = ack_message(message)
+    users = database_admin.get_all_users()
+    response = "There are no users" if len(users) == 0 else json.dumps(users)
+    pubsub_manager.publish(RESULT_TOPIC, response, message_id)
+
+
+def get_user_by_id_callback(message):
+    message_id = ack_message(message)
+    user_id = json.loads(message.data)["user_id"]
+    user = database_admin.get_user_by_id(user_id)
+    response = json.dumps(user) if user is not None else "User is not found"
+    pubsub_manager.publish(RESULT_TOPIC, response, message_id)
```

Comment:
Correct me if I wrong, but I didn't find a way to sender to know message id of message he just send
I've added artificial id in event service to solve that issue

Reply:
when you publish the message you can get the meessage id in [this way](https://github.com/Lohyna-international/lohfinder/blob/0c4a318f9ffaf75dc3d9090db2683f37e0bece72/services/user_service/pubsub.py#L33). But this will return message id only when the message is already published. So I use the following flow:

Sender will generate an id and put it to attributes. So when subscriber in user service receives the messages it uses this id to publish the result message. So responsibility of creating the ids for the message lays on client. I just use it.